src/codegen/abi.js

```diff
@@ -377,6 +377,16 @@ module.exports = class AbiCodeGenerator {
                       this.abi.name,
                       '',
                     ).name
+                  : output.get('type').substring(0, 6) === 'tuple['
```

Comment:
I'm not a big fan of matching only the beginning of the string. I'd rather do
```
output.get('type').match(/^tuple\[([0-9]+)?\]$/)
```
Same in the other places. This could be moved into an `isTupleArrayType(s)` helper.